src/main/java/Main.java

```diff
@@ -37,135 +43,176 @@
-            iloscNadrobek = s.nextInt();
+            amountOfOvertricks = scanner.nextInt();
         }
 
 
         System.out.println("Czy byla kontra? t-tak, n-nie");
-        kontra = s.next().charAt(0);
+        veto = scanner.next().charAt(0);
 
-        if (kontra == 't') {
+        if (veto == 't') {
             System.out.println("Czy byla rekontra? t-tak, n-nie");
-            rekontra = s.next().charAt(0);
-            if ((rekontra != 't') && (rekontra != 'n')) throw new RuntimeException("Podano niepoprawna litere");
-        } else if (kontra != 'n') throw new RuntimeException("Podano niepoprawna litere");
+            reveto = scanner.next().charAt(0);
+            if ((reveto != 't') && (reveto != 'n'))
+                throw new RuntimeException("Podano niepoprawna litere");
+        } else if (veto != 'n')
+            throw new RuntimeException("Podano niepoprawna litere");
 
 
     }
 
-    private int obliczIloscZdobytychPunktow() {
-        int kolumnaStartowa;
-        int rob = 0;
-        int wiersz = iloscWylicytowanychLew - 1;
-        int suma = 0;
-        int tab[][] = tab3;
-
-        if (etap.toLowerCase().equals("po")) tab = tab4;
+    private int calculateMatchScore() {
+        int startingCol;
+        int offset = 0;
+        int row = amountOfBidedTricks - 1;
+        int sum = 0;
+        int supplementaryTable[][] = tableOfGameValuesBeforeMatch;
+
+        if (stageOfTheMatch.toLowerCase().equals("po"))
+            supplementaryTable = tableOfGameValuesAfterMatch;
+
+        if (suit == Suit.TREFL || suit == Suit.KARO)
+            startingCol = 4;
+        else if (suit == Suit.KIER || suit == Suit.PIK)
+            startingCol = 7;
+        else startingCol = 10;
+
+        if (veto == 't') {
+            if (reveto == 't')
+                offset = 2;
+            else offset = 1;
+        }
 
-        if (kolor == Kolor.TREFL || kolor == Kolor.KARO) kolumnaStartowa = 4;
-        else if (kolor == Kolor.KIER || kolor == Kolor.PIK) kolumnaStartowa = 7;
-        else kolumnaStartowa = 10;
+        sum += supplementaryTable[row][startingCol + offset];
+        if (amountOfBloopers > 0)
+            sum -= amountOfBloopers * supplementaryTable[row][offset + 1];
+        else if (amountOfOvertricks > 0)
+            sum += amountOfOvertricks * supplementaryTable[MAX_AMOUNT_OF_TRICKS][startingCol + offset];
 
-        if (kontra == 't') {
-            if (rekontra == 't') rob = 2;
-            else rob = 1;
-        }
-        suma += tab[wiersz][kolumnaStartowa + rob];
-        if (iloscWpadek > 0) suma -= iloscWpadek * tab[wiersz][rob + 1];
-        else {
-            if (iloscNadrobek > 0) suma += iloscNadrobek * tab[MAX_ILOSC_LEW][kolumnaStartowa + rob];
 
-        }
+        return sum;
+    }
 
-        return suma;
+    private int getResultInInternationalMatchPoints() {
+        margin = calculateMatchScore() - calculateValueOfHand();
+        int i = 0;
+        int j = 0;
```

Comment:
Ta zmienna jest raczej niepotrzebna, bo potem jak się odwołujesz w kodzie i bierzesz tabela[][j] albo tabela[][j+1] to i tak nic nie wiadomo co to jest.